src/libs/ReportSecondaryActionUtils.ts

```diff
@@ -0,0 +1,332 @@
+
+    if (isExporter && isFinished && hasAccountingConnection && !syncEnabled) {
+        return true;
+    }
+
+    return false;
+}
+
+function isHoldAction(report: Report, reportTransactions: Transaction[]): boolean {
+    const isExpense = isExpenseReport(report);
+
+    if (!isExpense) {
+        return false;
+    }
+
+    const isOnHold = reportTransactions.some(isOnHoldTransactionUtils);
+
+    if (isOnHold) {
+        return false;
+    }
+
+    const isOpen = isOpenReport(report);
+    const isProcessing = isProcessingReport(report);
+    const isApproved = isReportApproved({report});
+
+    return isOpen || isProcessing || isApproved;
+}
+
+function isChangeWorkspaceAction(report: Report, policy: Policy, reportTransactions: Transaction[], violations: OnyxCollection<TransactionViolation[]>): boolean {
+    const isExpense = isExpenseReport(report);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const areWorkflowsEnabled = policy.areWorkflowsEnabled;
+    const isClosed = isClosedReport(report);
+
+    if (isExpense && isSubmitter && !areWorkflowsEnabled && isClosed) {
+        return true;
+    }
+
+    const isOpen = isOpenReport(report);
+    const isProcessing = isProcessingReport(report);
+
+    if (isSubmitter && (isOpen || isProcessing)) {
+        return true;
+    }
+
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+
+    if (isApprover && isProcessing) {
+        return true;
+    }
+
+    const isReportPayer = isPayer(getSession(), report, false, policy);
+    const isApproved = isReportApproved({report});
+
+    if (isReportPayer && (isApproved || isClosed)) {
+        return true;
+    }
+
+    const isAdmin = policy?.role === CONST.POLICY.ROLE.ADMIN;
+    const isReimbursed = isSettled(report);
+    const transactionIDs = reportTransactions.map((t) => t.transactionID);
+    const hasAllPendingRTERViolations = allHavePendingRTERViolation(transactionIDs, violations);
+
+    const shouldShowBrokenConnectionViolation = shouldShowBrokenConnectionViolationForMultipleTransactions(transactionIDs, report, policy, violations);
+
+    const userControlsReport = isSubmitter || isApprover || isAdmin;
+    const hasReceiptMatchViolation = hasAllPendingRTERViolations || (userControlsReport && shouldShowBrokenConnectionViolation);
+    const isReportExported = isExported(getReportActions(report));
+
+    if (isAdmin && ((!isReportExported && (isApproved || isReimbursed || isClosed)) || hasReceiptMatchViolation)) {
+        return true;
+    }
+
+    const isIOU = isIOUReport(report);
+    const hasOnlyPersonalWorkspace = hasNoPolicyOtherThanPersonalType();
+    const isReceiver = isReportManager(report);
+    if (isIOU && !hasOnlyPersonalWorkspace && isReceiver && isReimbursed) {
+        return true;
+    }
+
+    const isSender = isCurrentUserSubmitter(report.reportID);
+    // it's already satisified in line 215
+    if (isSender && isProcessing) {
+        return true;
+    }
```

Comment:
@luacmartins This condition is already satisfied in line 215: https://github.com/Expensify/App/blob/e45b47197b223a4a06ed200228f00aef777911c3/src/libs/ReportSecondaryActionUtils.ts#L212-L217

Aren't we missing in `isInvoice` in this condition? In other places `isSender` references Invoice report

Reply:
Hmm you're right. I think we can just get rid of this one, since the logic is the same for expense/invoice reports and AFAIK invoices can't be in Open state